File: 백준/Gold/1253. 좋다/좋다.py
import sys
n = int(input())
nums = list(map(int,sys.stdin.readline().split()))
answer = 0
nums.sort()

for i in range(n):
  target = nums[i]
  tmp_nums = nums[:i] + nums[i+1:]

  start, end = 0,n-2
  while(start<end):
    num = tmp_nums[start] + tmp_nums[end]
    if num == target:
      answer+=1
      break
    elif num > target:
      end-=1
    else:
      start+=1

print(answer)
    

File: 백준/Gold/1253. 좋다/좋다.js
const fs = require('fs')
const input = fs.readFileSync('/dev/stdin','utf8').trim().split('\n')
const N = Number(input.shift())
const values = input[0].trim().split(' ').map(Number)
let answer = 0
for (let [idx, value] of values.entries()){
    let lst = [...values]
    lst.splice(idx,1)

    for (let i = 0;i<N-1;i++){
        let flag = 0
        for (let j = i+1;j<N-1;j++){
            if (lst[i]+lst[j] === value) {
                flag =1
                break
            }
        }
        if (flag) {
            answer++
            break
        }
    }
}

console.log(answer)